hazelcast/src/main/java/com/hazelcast/nio/tcp/WriteHandler.java

```diff
@@ -172,7 +172,7 @@ private void unschedule() {
         // So the outputBuffer is empty, so we are going to unschedule ourselves.
         scheduled.set(false);
 
-        if (writeQueue.isEmpty() && urgentWriteQueue.isEmpty()) {
+        if (writeQueue.isEmpty() && urgentWriteQueue.isEmpty() && currentPacket == null) {
```

Comment:
I believe your changes in WriteHandler starts here 
https://github.com/hazelcast/hazelcast/commit/de557b4b1700a72f7a55f7a894cad95af689f24a
And version seems to be 3.5-SNAPSHOT. 
https://github.com/hazelcast/hazelcast/blob/de557b4b1700a72f7a55f7a894cad95af689f24a/pom.xml#L26